Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/AuditSearchBean.java

```diff
@@ -0,0 +1,269 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.SessionScoped;
+
+/**
+ *
+ * @author achidamb
+ */
+@ManagedBean(name = "auditSearchBean")
+@SessionScoped
+public class AuditSearchBean {
+
+    private String userId;
+    private Map<String, String> remoteHcidList;
+    private Date eventStartDate;
+    private Date eventEndDate;
+    private ArrayList<String> eventTypeList;
+    private int activeIndex = 0;
+    private String outcomeIndicator;
+    private ArrayList<String> selectedRemoteHcidList;
+    private ArrayList<String> selectedEventTypeList;
+    private Map<String, String> eventOutcomeIndicatorList;
+    private ArrayList<Audit> auditRecordList;
+    private String auditMessage;
+    private boolean auditFound;
+    private String messageId;
+    private String relatesTo;
+
+    public AuditSearchBean() {
+        setEventTypeList(populateEventTypeList());
+        setRemoteHcidList(populateRemoteOrgHcid());
+        setEventOutcomeIndicatorList(populateEventOutcomeIndicatorValues());
+    }
+
+    public void searchAudit() {
+        AuditService impl = new AuditServiceImpl();
+        this.auditRecordList = impl.createMockAuditRecord();
+        if (this.auditRecordList.isEmpty() && this.auditRecordList == null) {
+            this.auditMessage = "Audit Records not found";
+            auditFound = false;
+        } else {
+            this.auditMessage = "Audit Records Found";
+            auditFound = true;
+        }
+    }
+
+    public void searchAuditMessageId() {
+        AuditServiceImpl impl = new AuditServiceImpl();
+        this.auditRecordList = impl.createMockAuditRecord();
+        if (this.auditRecordList.isEmpty() && this.auditRecordList == null) {
+            this.auditMessage = "Audit Records not found";
+            auditFound = false;
+        } else {
+            this.auditMessage = "Audit Records Found";
+            auditFound = true;
+        }
+    }
+
+    public String clearAuditTab() {
+        this.eventEndDate = null;
+        this.eventStartDate = null;
+        if (this.selectedEventTypeList != null && !this.selectedEventTypeList.isEmpty()) {
+            this.selectedEventTypeList.clear();
+        }
+        if (this.selectedRemoteHcidList != null && !this.selectedRemoteHcidList.isEmpty()) {
+            this.selectedRemoteHcidList.clear();
+        }
+        this.outcomeIndicator = null;
+        this.userId = null;
+        if (this.auditRecordList != null && !auditRecordList.isEmpty()) {
+            this.auditRecordList.clear();
+        }
+        this.auditMessage = "";
+        this.auditFound = false;
+        return NavigationConstant.AUDIT_SEARCH_PAGE;
+    }
+
+    public String clearAuditTabMessageId() {
+        this.relatesTo = null;
+        this.messageId = null;
+        if (this.auditRecordList != null && !auditRecordList.isEmpty()) {
```

Comment:
Checking for Not null is sufficient here. We don't need to check !auditRecordList.isEmpty() as we are clearing the list.